modules/azure/compute.go

```diff
@@ -8,97 +8,392 @@ import (
+func GetVirtualMachineTagsE(vmName string, resGroupName string, subscriptionID string) (map[string]string, error) {
 	// Setup a blank map to populate and return
-	tags := make(map[string]string)
+	var tags map[string]string
 
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Range through existing tags and populate above map accordingly
+	for k, v := range vm.Tags {
+		tags[k] = *v
+	}
+
+	return tags, nil
+}
+
+// ***************************************************** //
+// Get multiple Virtual Machines from a Resource Group
+// ***************************************************** //
+
+// ListVirtualMachinesForResourceGroup gets a list of all Virtual Machine names in the specified Resource Group.
+// This function would fail the test if there is an error.
+func ListVirtualMachinesForResourceGroup(t testing.TestingT, resGroupName string, subscriptionID string) []string {
+	vms, err := ListVirtualMachinesForResourceGroupE(resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return vms
+}
+
+// ListVirtualMachinesForResourceGroupE gets a list of all Virtual Machine names in the specified Resource Group.
+func ListVirtualMachinesForResourceGroupE(resourceGroupName string, subscriptionID string) ([]string, error) {
+	var vmDetails []string
+
+	vmClient, err := GetVirtualMachineClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	vms, err := vmClient.List(context.Background(), resourceGroupName)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, v := range vms.Values() {
+		vmDetails = append(vmDetails, *v.Name)
+	}
+	return vmDetails, nil
+}
+
+// GetVirtualMachinesForResourceGroup gets all Virtual Machine objects in the specified Resource Group. Each
+// VM Object represents the entire set of VM compute properties accessible by using the VM name as the map key.
+// This function would fail the test if there is an error.
+func GetVirtualMachinesForResourceGroup(t testing.TestingT, resGroupName string, subscriptionID string) map[string]compute.VirtualMachineProperties {
+	vms, err := GetVirtualMachinesForResourceGroupE(resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return vms
+}
+
+// GetVirtualMachinesForResourceGroupE gets all Virtual Machine objects in the specified Resource Group. Each
+// VM Object represents the entire set of VM compute properties accessible by using the VM name as the map key.
+func GetVirtualMachinesForResourceGroupE(resourceGroupName string, subscriptionID string) (map[string]compute.VirtualMachineProperties, error) {
+	vmClient, err := GetVirtualMachineClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	vms, err := vmClient.List(context.Background(), resourceGroupName)
+	if err != nil {
+		return nil, err
+	}
+
+	vmDetails := make(map[string]compute.VirtualMachineProperties, len(vms.Values()))
+	for _, v := range vms.Values() {
+		vmDetails[string(*v.Name)] = *v.VirtualMachineProperties
```

Comment:
You can guess what I'm going to ask 😁 

Also, why does `*v.Name` need to be cast to a `string`?

Reply:
removed unnecessary cast!